uplink/converters/pydantic_v1.py

```diff
@@ -0,0 +1,113 @@
+"""
+This module defines a converter that uses :py:mod:`pydantic.v1` models
+to deserialize and serialize values.
+"""
+
+from uplink.converters import register_default_converter_factory
+from uplink.converters.interfaces import Factory, Converter
+from uplink.utils import is_subclass
+
+
+def _encode_pydantic_v1(obj):
+    from pydantic.v1.json import pydantic_encoder
+
+    # json atoms
+    if isinstance(obj, (str, int, float, bool)) or obj is None:
+        return obj
+
+    # json containers
+    if isinstance(obj, dict):
+        return {_encode_pydantic_v1(k): _encode_pydantic_v1(v) for k, v in obj.items()}
+    if isinstance(obj, list):
+        return [_encode_pydantic_v1(i) for i in obj]
+
+    # pydantic v1 types
+    return _encode_pydantic_v1(pydantic_encoder(obj))
+
+
+class _PydanticV1RequestBody(Converter):
+    def __init__(self, model):
+        self._model = model
+
+    def convert(self, value):
+        if isinstance(value, self._model):
+            return _encode_pydantic_v1(value)
+        return _encode_pydantic_v1(self._model.parse_obj(value))
+
+
+class _PydanticV1ResponseBody(Converter):
+    def __init__(self, model):
+        self._model = model
+
+    def convert(self, response):
+        try:
+            data = response.json()
+        except AttributeError:
+            data = response
+
+        return self._model.parse_obj(data)
+
+
+class PydanticV1Converter(Factory):
```

Comment:
Would it be too complicated to handle both v1 an v2 models in a single factory? I think keeping them separate also makes sense, just thinking out loud. 

For example, a "unified" `PydanticConverter` instance could create the appropriate request and response body converters based on whether the model is a `v1` or `v2`.

Reply:
:heavy_check_mark: